heartbeat/monitors/task.go

```diff
@@ -133,10 +133,15 @@ func (t *configuredJob) makeSchedulerTaskFunc() scheduler.TaskFunc {
 func (t *configuredJob) Start() {
 	var err error
 
+	fields := common.MapStr{"event": common.MapStr{"dataset": "uptime"}}
+	if t.monitor.factoryMetadata != nil {
+		fields.DeepUpdate(t.monitor.factoryMetadata.Get())
+	}
+
 	t.client, err = t.monitor.pipelineConnector.ConnectWith(beat.ClientConfig{
 		EventMetadata: t.config.EventMetadata,
 		Processor:     t.processors,
-		Fields:        common.MapStr{"event": common.MapStr{"dataset": "uptime"}},
+		Fields:        fields,
```

Comment:
there is a `DynamicFields` here in `ClientConfig`, were you can pass t.monitor.factoryMetadata directly, no need for that `DeepUpdate`